Copying an Exponential Histogram results in nil pointer dereference

### Component(s)

processor/metricstransform

### What happened?

## Description
Attempting to perform any operation that copies an exponential histogram will result in a nil pointer dereference error as the code currently calls the incorrect method on the metric. 

## Steps to Reproduce
Have a config that attempts an operation on an exponential histogram metric that results in a copy, such as `combine`

## Expected Result
The metric is correctly processed and exported.

## Actual Result
A nil pointer dereference error is printed to the logs. 

### Collector version

v0.86.0

### Environment information

## Environment
All



### OpenTelemetry Collector configuration

```yaml
processors:
  metricstransform:
    transforms:
      - include: ^<regex_for_exponential_histogram_metric>\.(?P<new_label>.+)$
        match_type: regexp
        action: combine
        new_name: new_metric
```


### Log output

```shell
service/service.go:161	Everything is ready. Begin running and processing data.
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0x2287efb]

goroutine 51 [running]:
go.opentelemetry.io/collector/pdata/pmetric.Histogram.AggregationTemporality(...)
	/go/pkg/mod/go.opentelemetry.io/collector/pdata@v1.0.0-rcv0014/pmetric/generated_histogram.go:45
```


### Additional context

_No response_

--- a/.chloggen/fix-metricstransformprocessor-copy-exponential-histogram.yaml
+++ b/.chloggen/fix-metricstransformprocessor-copy-exponential-histogram.yaml
@@ -1,0 +1,27 @@
+# Use this changelog template to create an entry for release notes.
+
+# One of 'breaking', 'deprecation', 'new_component', 'enhancement', 'bug_fix'
+change_type: 'bug_fix'
+
+# The name of the component, or a single word describing the area of concern, (e.g. filelogreceiver)
+component: 'metricstransformprocessor'
+
+# A brief description of the change.  Surround your text with quotes ("") if it needs to start with a backtick (`).
+note: 'Fixes a nil pointer dereference when copying an exponential histogram'
+
+# Mandatory: One or more tracking issues related to the change. You can use the PR number here if no issue exists.
+issues: [27409]
+
+# (Optional) One or more lines of additional information to render under the primary note.
+# These lines will be padded with 2 spaces and then inserted directly into the document.
+# Use pipe (|) for multiline entries.
+subtext:
+
+# If your change doesn't affect end users or the exported elements of any package,
+# you should instead start your pull request title with [chore] or use the "Skip Changelog" label.
+# Optional: The change log or logs in which this entry should be included.
+# e.g. '[user]' or '[user, api]'
+# Include 'user' if the change is relevant to end users.
+# Include 'api' if there is a change to a library API.
+# Default: '[user]'
+change_logs: [user]

--- a/processor/metricstransformprocessor/metrics_transform_processor_otlp.go
+++ b/processor/metricstransformprocessor/metrics_transform_processor_otlp.go
@@ -454,7 +454,7 @@ func copyMetricDetails(from, to pmetric.Metric) {
 	case pmetric.MetricTypeHistogram:
 		to.SetEmptyHistogram().SetAggregationTemporality(from.Histogram().AggregationTemporality())
 	case pmetric.MetricTypeExponentialHistogram:
-		to.SetEmptyExponentialHistogram().SetAggregationTemporality(from.Histogram().AggregationTemporality())
+		to.SetEmptyExponentialHistogram().SetAggregationTemporality(from.ExponentialHistogram().AggregationTemporality())
 	case pmetric.MetricTypeSummary:
 		to.SetEmptySummary()
 	}

--- a/processor/metricstransformprocessor/metrics_transform_processor_testcases_test.go
+++ b/processor/metricstransformprocessor/metrics_transform_processor_testcases_test.go
@@ -1162,6 +1162,23 @@ var (
 			},
 		},
 		{
+			name: "combine_exponential_histogram",
+			transforms: []internalTransform{
+				{
+					MetricIncludeFilter: internalFilterRegexp{include: regexp.MustCompile("^metric(?P<namedsubmatch>[12])$")},
+					Action:              Combine,
+					NewName:             "new",
+				},
+			},
+			in: []pmetric.Metric{
+				metricBuilder(pmetric.MetricTypeExponentialHistogram, "metric1").build(),
+				metricBuilder(pmetric.MetricTypeExponentialHistogram, "metric2").build(),
+			},
+			out: []pmetric.Metric{
+				metricBuilder(pmetric.MetricTypeExponentialHistogram, "new", "namedsubmatch").build(),
+			},
+		},
+		{
 			name: "combine_error_type",
 			transforms: []internalTransform{
 				{

--- a/processor/metricstransformprocessor/metrics_transform_processor_testcases_test.go
+++ b/processor/metricstransformprocessor/metrics_transform_processor_testcases_test.go
@@ -1165,9 +1165,11 @@ var (
 			name: "combine_exponential_histogram",
 			transforms: []internalTransform{
 				{
-					MetricIncludeFilter: internalFilterRegexp{include: regexp.MustCompile("^metric(?P<namedsubmatch>[12])$")},
-					Action:              Combine,
-					NewName:             "new",
+					MetricIncludeFilter: internalFilterRegexp{
+						include: regexp.MustCompile("^metric(?P<namedsubmatch>[12])$"),
+					},
+					Action:  Combine,
+					NewName: "new",
 				},
 			},
 			in: []pmetric.Metric{

